src/core/meshcop/meshcop_tlvs.hpp

```diff
@@ -337,7 +337,10 @@ class NetworkNameTlv: public Tlv
      * @retval FALSE  If the TLV does not appear to be well-formed.
      *
      */
-    bool IsValid(void) const { return GetLength() <= sizeof(*this) - sizeof(Tlv); }
+    bool IsValid(void) const
+    {
+        return GetLength() <= sizeof(*this) - sizeof(Tlv) && GetLength() <= OT_NETWORK_NAME_MAX_SIZE;
```

Comment:
Is this additional check redundant?  I believe `sizeof(*this) - sizeof(Tlv)` should equal `OT_NETWORK_NAME_MAX_SIZE`.